lib/bolt/config.rb

```diff
@@ -248,6 +254,7 @@ def update(options)
     def load_file(path)
       data = Bolt::Util.read_config_file(path, [default_config], 'config')
       update_from_file(data) if data
+      validate_hiera_conf(data ? data['hiera-config'] : nil)
```

Comment:
Idea here is if `hiera-config` is specified in `configfile` or exists in default location then make sure it can be parsed (with the expected bolt read_config_file logging). 

Validation that hiera v5 is executed only if there is an apply block in a plan. That way we do not error out if a hiera.yaml file is laying around somewhere but the `apply` is not used in a plan.

Reply:
That makes sense.